contracts/modules/LoanSettings/LoanSettings.sol

```diff
@@ -91,178 +68,111 @@ contract LoanSettings is State, InterestHandler, LoanSettingsEvents, PausableGua
-            );
+            emit LoanParamsIdDisabled(loanParamsLocal.id, loanParamsLocal.owner);
         }
     }
 
-    function getLoanParams(
-        bytes32[] memory loanParamsIdList)
-        public
-        view
-        returns (LoanParams[] memory loanParamsList)
-    {
-        loanParamsList = new LoanParams[](loanParamsIdList.length);
-        uint256 itemCount;
-
-        for (uint256 i = 0; i < loanParamsIdList.length; i++) {
-            LoanParams memory loanParamsLocal = loanParams[loanParamsIdList[i]];
-            if (loanParamsLocal.id == 0) {
-                continue;
-            }
-            loanParamsList[itemCount] = loanParamsLocal;
-            itemCount++;
-        }
-
-        if (itemCount < loanParamsList.length) {
-            assembly {
-                mstore(loanParamsList, itemCount)
-            }
+    function modifyLoanParams(LoanParams[] calldata loanParamsList) external onlyGuardian {
+        for (uint256 i = 0; i < loanParamsList.length; i++) {
+            require(
+                supportedTokens[loanParamsList[i].loanToken] &&
+                    supportedTokens[loanParamsList[i].collateralToken] &&
+                    loanParamsList[i].id ==
+                    generateLoanParamId(
+                        loanParamsList[i].loanToken,
+                        loanParamsList[i].collateralToken,
+                        loanParamsList[i].maxLoanTerm == 0 // isTorqueLoan
+                            ? true
+                            : false && loanParamsList[i].minInitialMargin > loanParamsList[i].maintenanceMargin
+                    ),
+                "invalid loanParam"
+            );
+            emit LoanParamsSetup(
+                loanParamsList[i].id,
+                loanParamsList[i].owner,
+                loanParamsList[i].loanToken,
+                loanParamsList[i].collateralToken,
+                loanParamsList[i].minInitialMargin,
+                loanParamsList[i].maintenanceMargin,
+                loanParamsList[i].maxLoanTerm
+            );
+            emit LoanParamsIdSetup(loanParamsList[i].id, loanParamsList[i].owner);
         }
     }
 
-    function getLoanParamsList(
+    function migrateLoanParamsList(
         address owner,
         uint256 start,
-        uint256 count)
-        external
-        view
-        returns (bytes32[] memory loanParamsList)
-    {
+        uint256 count
+    ) external onlyGuardian {
         EnumerableBytes32Set.Bytes32Set storage set = userLoanParamSets[owner];
         uint256 end = start.add(count).min256(set.length());
         if (start >= end) {
-            return loanParamsList;
+            return;
         }
-        count = end-start;
-
-        loanParamsList = new bytes32[](count);
-        for (uint256 i = --end; i >= start; i--) {
-            loanParamsList[--count] = set.get(i);
-
-            if (i == 0) {
-                break;
-            }
+        count = end - start;
```

Comment:
can be removed as count is unused from this point forward.

Reply:
fixed